src/bootstrap/builder.rs

```diff
@@ -2043,6 +2043,13 @@ impl<'a> Builder<'a> {
             rustflags.arg("-Zinline-mir");
         }
 
+        // set rustc args passed from command line
+        let rustc_args =
+            self.config.cmd.rustc_args().iter().map(|s| s.to_string()).collect::<Vec<_>>();
+        if !rustc_args.is_empty() {
+            cargo.env("RUSTFLAGS", &rustc_args.join(" "));
+        }
```

Comment:
I don't understand this patch, it doesn't make sense to me. This is setting the RUSTFLAGS env var, but when this `Cargo` instance is later turned into a `Command`, the RUSTFLAGS env var is anyway overwritten again:

https://github.com/rust-lang/rust/blob/c350d3c5dd00ed0727ffe72fc33637d8c7537733/src/bootstrap/src/core/builder.rs#L2414-L2418

Reply:
Shouldn't this be added to `rustflags` instead?

Reply:
Also the --help text says this should only be passed when running tests; I don't see that reflected in the code.

Reply:
Yeah, this isn't correct and fixed in https://github.com/rust-lang/rust/pull/123489.